web/src/user_status_ui.ts

```diff
@@ -46,18 +49,22 @@ export function open_user_status_modal() {
     });
 }
 
-export function submit_new_status() {
+export function submit_new_status(): void {
     const user_id = people.my_current_user_id();
     let old_status_text = user_status.get_status_text(user_id) ?? "";
     old_status_text = old_status_text.trim();
-    const old_emoji_info = user_status.get_status_emoji(user_id) || {};
-    const new_status_text = input_field().val().trim();
-
+    const old_emoji_info = user_status.get_status_emoji_info_to_compare(user_id);
+    let new_status_text = input_field().val();
+    assert(typeof new_status_text === "string");
+    new_status_text = new_status_text.trim();
     if (
         old_status_text === new_status_text &&
-        old_emoji_info.emoji_name === selected_emoji_info.emoji_name &&
-        old_emoji_info.reaction_type === selected_emoji_info.reaction_type &&
-        old_emoji_info.emoji_code === selected_emoji_info.emoji_code
+        ((old_emoji_info.type === "emoji_info_present" &&
+            old_emoji_info.emoji_name === selected_emoji_info.emoji_name &&
+            old_emoji_info.reaction_type === selected_emoji_info.reaction_type &&
+            old_emoji_info.emoji_code === selected_emoji_info.emoji_code) ||
+            (old_emoji_info.type === "emoji_info_absent" &&
+                Object.keys(selected_emoji_info).length === 0))
```

Comment:
This checks if the `selected_emoji_info` in an empty object ie. user hasn't set any emoji status.

Reply:
I think probably it's worth turning this into a function; giant if statements like this are hard to read, and I think concretely, if this check question is well-expressed as `emoji_status_fields_changed(old_emoji_info, selected_emoji_info)`, probably having a function with that name would lead to much more readable code.

Reply:
Thanks!. Added a prep commit .